src/python/pants/engine/scheduler.py

```diff
@@ -297,9 +297,16 @@ def with_fork_context(self, func):
 
   def _run_and_return_roots(self, session, execution_request):
     raw_roots = self._native.lib.scheduler_execute(self._scheduler, session, execution_request)
+    remaining_runtime_exceptions_to_capture = deque(self._native.cffi_extern_method_runtime_exceptions())
```

Comment:
This `deque` is not imported, which causes an exception to be thrown from this code itself

Reply:
In addition, its local to the stack so only one thread can use it and then only to `append` and `__bool__`; so a list will do.

Reply:
I was unfamiliar with the `.pop(<index>)` method on `list`s! I have made this change.